src/promptflow/promptflow/_core/run_tracker.py

```diff
@@ -360,6 +360,25 @@ def get_run(self, run_id):
     def persist_node_run(self, run_info: RunInfo):
         self._storage.persist_node_run(run_info)
 
+    def persist_selected_node_runs(self, run_info: FlowRunInfo, node_names: List[str]):
+        """Persists the node runs for the specified node names.
+
+        Args:
+            run_info (FlowRunInfo): The flow run information.
+            node_names (List[str]): The names of the nodes to persist.
+
+        Returns:
+            None
+        """
+        run_id = run_info.run_id
+        selected_node_run_info = [
```

Comment:
We can use `(` to replace `[` here to make the result to be a generator instead of a list.

Reply:
What's the benefit of using generator? For performance consideration, right?